lib/plugin/instance.rb

```diff
@@ -1018,6 +1018,21 @@ def add_topic_static_page(page, options = {}, &blk)
     StaticController::CUSTOM_PAGES[page] = blk ? { topic_id: blk } : options
   end
 
+  # Lets plugin define custom unsubscribe keys,
+  # set custom instance variables on the controller's unsubscribe action,
+  # and describe what unsubscribing for that key does.
+  #
+  # The method receives a class that inherits from `Email::BaseEmailUnsubscriber`.
+  # Take a look at it to know how to implement your child class.
+  #
+  # In conjunction with this, you have to define a connector under app/views/connectors/unsubscribe_options.
```

Comment:
Can we elaborate more here as well on what the connectors should look like? Alternatively, is there a possibility we can generate the HTML based on the key instead of having to add a connector?